### PR TITLE
Smart debug tests

### PR DESCRIPTION
**Pull request by aarondr77**
# Description

First five smart debug tests! 

- Creates the `SmartDebugTestCase` and `DebugPromptGenerator`
- Creates shared utilities between the chat and debug test runners
- Creates a new assert directory where we can add more asserts (ie: maybe we want to check that the ai message contains the words "restart kernel"
- Let's us check the outputs of code. ie: we can check that `print('hello world)` is fixed and actually prints the message. 

# Testing

Run the new tests + run the old tests to make sure the refactor didn't break anything. 

# Documentation

No.

### CODE REVIEW

File: evals/asserts/equal_outputs.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+def assert_equal_outputs(expected_output: str, actual_output: str) -> bool:
```
Comment (aarondr77) [inline evals/asserts/equal_outputs.py:1]: Add a docstring

File: evals/main.py [hunk 2]
```diff
@@ -1,91 +1,23 @@
...
-from evals.utils import are_globals_equal, get_globals_to_compare, get_script_from_cells, print_test_case_result_tables
-from evals.test_cases import *
+from evals.test_runners.code_gen_test_runner import run_code_gen_tests
+from evals.test_runners.smart_debugger_test_runner import run_smart_debug_tests
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description='Run evaluation tests')
+        
+    # Require the user to specify if they want to run chat or smart debug tests
+    parser.add_argument('--test-type', type=str, required=True, choices=['chat', 'smart_debug'], help='Type of tests to run (chat or smart_debug)')
```
Comment (aarondr77) [inline evals/main.py:9]: Update the README

File: evals/test_runners/code_gen_test_runner.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    try:
+        exec(expected_code, expected_globals)
+        exec(actual_code, actual_globals)
+    except Exception as e:
+        # Fail early if we can't execute the code
+        print("Test Failed: ")
+        print(f"Expected code:\n{expected_code}")
+        print(f"\nActual code:\n{actual_code}")
+        print(f"Error: {e}")
```
Comment (aarondr77) [inline evals/test_runners/code_gen_test_runner.py:76]: Move this to the new helper function

File: evals/test_runners/smart_debugger_test_runner.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    # Get the expected code script 
+    expected_code = script_without_invalid_code + "\n" + test.correct_code
+
+    # TODO: Turn this into a function and add it to utils.py
+    # So that we can compare the results of the two scripts, create global context for 
+    # each script. When calling exec, the globals are updated in place.
+
+    expected_globals = {}
+    actual_globals = {}
```
Comment (aarondr77) [inline evals/test_runners/smart_debugger_test_runner.py:92]: Delete

**Review by aarondr77 — COMMENTED** [review]
small cleanup

File: evals/main.py
Comment (ngafar) [inline evals/main.py:1]: Great changes here! Super easy to hop into, and easily know where to go next.

**Review by ngafar — APPROVED** [review]

**Pull request merged**